exporters/prometheus/src/main/java/io/opentelemetry/exporters/prometheus/MetricAdapter.java

```diff
@@ -105,10 +105,18 @@ static Type toMetricFamilyType(MetricData.Descriptor.Type type) {
     if (descriptor.getConstantLabels().size() != 0) {
       constLabelNames = new ArrayList<>(descriptor.getConstantLabels().size());
       constLabelValues = new ArrayList<>(descriptor.getConstantLabels().size());
-      for (Map.Entry<String, String> entry : descriptor.getConstantLabels().entrySet()) {
-        constLabelNames.add(toLabelName(entry.getKey()));
-        constLabelValues.add(entry.getValue() == null ? "" : entry.getValue());
-      }
+      final List<String> finalConstLabelNames = constLabelNames;
+      final List<String> finalConstLabelValues = constLabelValues;
+      descriptor
+          .getConstantLabels()
+          .forEach(
+              new KeyValueConsumer<String>() {
+                @Override
+                public void consume(String key, String value) {
+                  finalConstLabelNames.add(toLabelName(key));
+                  finalConstLabelValues.add(value == null ? "" : value);
+                }
+              });
```

Comment:
Please create an issue to revisit this logic and assign to me. seems we do unnecessary work.

Reply:
https://github.com/open-telemetry/opentelemetry-java/issues/1341